src/coordination/include/coordination/coordinator_communication_config.hpp

```diff
@@ -56,6 +57,33 @@ struct CoordinatorInstanceInitConfig {
   }
 };
 
+struct CoordinatorStateManagerConfig {
+  uint32_t coordinator_id_{0};
+  int coordinator_port_{0};
+  int bolt_port_{0};
+  std::filesystem::path state_manager_durability_dir_;
+  std::shared_ptr<kvstore::KVStore> durability_store_;
+
+  CoordinatorStateManagerConfig(uint32_t coordinator_id, int coordinator_port, int bolt_port,
+                                std::filesystem::path state_manager_durability_dir,
+                                std::shared_ptr<kvstore::KVStore> durability_store)
+      : coordinator_id_(coordinator_id),
+        coordinator_port_(coordinator_port),
+        bolt_port_(bolt_port),
+        state_manager_durability_dir_(std::move(state_manager_durability_dir)),
```

Comment:
Why do you need durability_dir_ for StateManagerConfig and not for StateMachineConfig?

Reply:
For the state manager, I decided that we need always durability, and that is why we are passing durability dir, whereas for the state machine and log store we only use durability if the user says so.

Reply:
resolved